articles/introduction-to-knn-algorithm/index.md

```diff
@@ -0,0 +1,232 @@
+title: An Introduction to KNN Algorithm
+description: In this article, we have covered the basics of a supervised machine learning algorithm called K-Nearest Neighbors. We will also learn to build this algorithm from scratch.
+author: srishilesh-p-s
+date: 2021-04-04T00:00:00-20:00
+topics: []
+excerpt_separator: <!--more-->
+images:
+  - url: /engineering-education/introduction-to-knn-algorithm/hero.jpg
+    alt: An Introduction to KNN Algorithm
+---
+In this article, we will learn about a supervised machine learning algorithm called K-Nearest Neighbors (KNN). This algorithm can be used for tasks like classification and prediction. By the end of this article, you will get an overview of various supervised machine learning algorithms, what the KNN algorithm is, how it works, and also learn to build the algorithm from scratch.
+<!--more-->
+As a prerequisite, a little understanding of machine learning and Python would help the beginners.
+
+### Table of contents
+- [Supervised learning](#supervised-learning)
+- [K-Nearest Neighbors algorithm](#k-nearest-neighbors-algorithm)
+- [Step by step guide](#step-by-step-guide)
+- [Conclusion](#conclusion)
+- [Further reading](#further-reading)
+
+### Supervised learning
+> According to [Wikipedia](https://en.wikipedia.org/wiki/Supervised_learning), Supervised learning is the machine learning task of learning a function that maps an input to an output based on example input-output pairs. It infers a function from labeled training data consisting of a set of training examples.
+
+In simple words, it can be said that we are teaching the machine how to predict or classify by showing some samples (training and testing) of the possible predictions or classifications.
+
+We use train data with features and labels, to enable the machine to learn first. Later, it is being validated or tested using the test data. Thereby, this method of learning is called supervised learning.
+
+### K-Nearest Neighbors algorithm
+K-Nearest Neighbors (KNN) algorithm is one such supervised learning that can be used for classification and regression.
+
+Steps followed in KNN algorithm:
+1. Load the train and test datasets.
+2. Specify or choose the value of `K`.
+3. For each point on the test data perform the following:
+   3.1. Calculate the distance between the point and each point of the training dataset. We can use Euclidean distance or Manhattan distance.
+   3.2. Sort the values in ascending order based on distances.
+   3.3. Find the top K values from the sorted list.
+   3.4. Find the frequency (Mode) of the labels of the top K values.
+   3.5. Assign the mode to the test data point.
+   3.6. The assigned value is the classified or predicted value for the particular test data point.
+
+![KNN algorithm](/engineering-education/introduction-to-knn-algorithm/knn.png)
+*[Image source](https://www.javatpoint.com/k-nearest-neighbor-algorithm-for-machine-learning): KNN algorithm shown visually*
+
```

Comment:
It would be great if you could include a link to an article that explains KNN mathematically in detail